### PR TITLE
SAPP-1099: NewRelic support

### PR DESCRIPTION
**Pull request by acbramley**
Adds NewRelic field and environment variables.

To test, add a `/code/private/SHEPHERD_NEW_RELIC_LICENSE` file locally with a license key as the contents.

Sample running locally using the PNX NewRelic account:
Screen Shot 2020-11-13 at 12 10 17 pm

### CODE REVIEW

File: web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module [hunk 1]
```diff
@@ -125,5 +121,14 @@ function shp_orchestration_shp_env_vars(NodeInterface $environment) {
     'WEB_PATH' => $environment->field_shp_path->value,
   ];
 
+  // Add NewRelic environment variables if enabled.
+  if ($environment->field_newrelic_enabled->value) {
+    $env_vars['NEW_RELIC_ENABLED'] = 'true';
+    // @TODO Figure out how the license key is passed in - probably through
+    // shepherd secrets?
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module:128]: Doesn't it need to be in the .ini file?

File: web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module [hunk 2]
```diff
@@ -110,11 +110,7 @@ function shp_orchestration_shp_project_form_validate(array $form, FormStateInter
  */
 function shp_orchestration_shp_env_vars(NodeInterface $environment) {
   /** @var Drupal\Node\NodeInterface $site */
-  $site = $environment->field_shp_site
-    ->first()
-    ->get('entity')
-    ->getTarget()
-    ->getValue();
+  $site = $environment->field_shp_site->entity;
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module:113]: Bueno. I reckon I wrote it the old way.

File: web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module [hunk 3]
```diff
@@ -125,5 +121,14 @@ function shp_orchestration_shp_env_vars(NodeInterface $environment) {
     'WEB_PATH' => $environment->field_shp_path->value,
   ];
 
+  // Add NewRelic environment variables if enabled.
+  if ($environment->field_newrelic_enabled->value) {
+    $env_vars['NEW_RELIC_ENABLED'] = 'true';
+    // @TODO Figure out how the license key is passed in - probably through
+    // shepherd secrets?
+    $env_vars['NEW_RELIC_LICENSE_KEY'] = '';
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module:129]: Do we want this settable at project level ? and environment level ?

File: web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module (hunk 1 above)
Comment (acbramley) [reply web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module:128]: The ini is pulling it through from this environment variable

File: web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module (hunk 3 above)
Comment (acbramley) [reply web/modules/custom/shepherd/shp_orchestration/shp_orchestration.module:129]: It should most likely come from a secret.

**Comment by pingers** [comment]
@acbramley Have we jumped the shark with the config and secrets madness?

**Review by pingers — APPROVED** [review]
Ship it! :ship:

**Pull request merged**